src/components/projecteditor/control/control.js

```diff
@@ -860,13 +860,35 @@ export default class Control extends Component {
         e.preventDefault();
         e.stopPropagation();
 
-        var name;
-        name = prompt("Please give the contract a name:");
-        if (!name) return;
-        if(!name.match(/^([a-zA-Z0-9-_]+)$/) || name.length > 16) {
-            alert('Illegal contract name. Only A-Za-z0-9, dash (-) and underscore (_) allowed. Max 16 characters.');
-            return;
+        var name = undefined;
+        do {
+            if(name === undefined ){
+                name = prompt('Please give the contract a name:', "Only 16 character Allowed");
+            }
+            else if (name !== null && name.length > 16) {
+                name = prompt('Only 16 character Allowed:', name);
+            }
+            else if(!name){
+                return null;
+            }
+            else if(!name.match(/^([a-zA-Z0-9-_]+)$/)) {
```

Comment:
This condition likely never gets executed, considering `name` can be set in the very first `if`, rendering the loop condition `false` for the next iteration.

Reply:
yeah, there was a problem that is fixed now.